### PR TITLE
Fix rust compile warning and an obvious print error in array.rs

### PR DESCRIPTION
**Pull request by rongyi**
fix rust compile warning and array print error and some linked list error, jus reading the rust code, will try my best to make rust code more readable and more elegantly.

### CODE REVIEW

File: codes/rust/chapter_array_and_linkedlist/linked_list.rs [hunk 1]
```diff
@@ -26,9 +26,9 @@ pub fn remove<T>(n0: &Rc<RefCell<ListNode<T>>>) {
     };
     // n0 -> P -> n1
     let P = n0.borrow_mut().next.take();
-    if let Some(node) = P {
-        let n1 = node.borrow_mut().next.take();
```
Comment (krahets) [inline codes/rust/chapter_array_and_linkedlist/linked_list.rs:30]: please do not modify the variable names. They are consistent across different programming languages.

File: codes/rust/chapter_array_and_linkedlist/linked_list.rs [hunk 2]
```diff
@@ -73,7 +74,7 @@ fn main() {
 
     /* 插入节点 */
     insert(&n0, ListNode::new(0));
-    print!("插入节点后的链表为 ");
+    print!("插入节点0后的链表为 ");
```
Comment (krahets) [inline codes/rust/chapter_array_and_linkedlist/linked_list.rs:77]: Do not modify the comments. They are consistent across different programming languages.

File: codes/rust/include/list_node.rs [hunk 3]
```diff
@@ -4,9 +4,9 @@
  * Author: codingonion (coderonion@gmail.com)
```
Comment (krahets) [inline codes/rust/include/list_node.rs:4]: Plz add your info here

File: codes/rust/chapter_array_and_linkedlist/linked_list.rs (hunk 1 above)
Comment (diucicd) [reply codes/rust/chapter_array_and_linkedlist/linked_list.rs:30]: oh, got you

File: codes/rust/chapter_array_and_linkedlist/linked_list.rs (hunk 2 above)
Comment (diucicd) [reply codes/rust/chapter_array_and_linkedlist/linked_list.rs:77]: got you

File: codes/rust/include/list_node.rs (hunk 3 above)
Comment (diucicd) [reply codes/rust/include/list_node.rs:4]: so considerate!

**Review by krahets — APPROVED** [review]
Great changes. Thank you!

**Pull request merged**